evm_loader/lib/src/commands/trace.rs

```diff
@@ -27,26 +29,26 @@ pub async fn trace_transaction(
     solana_accounts: &[Pubkey],
     trace_call_config: TraceCallConfig,
 ) -> Result<TracedCall, NeonError> {
-    let mut tracer = Tracer::new(trace_call_config.trace_config.enable_return_data);
-
-    let (emulation_result, _storage) = evm_loader::evm::tracing::using(&mut tracer, || async {
-        emulate_transaction(
-            rpc_client,
-            evm_loader,
-            tx,
-            token,
-            chain_id,
-            steps,
-            commitment,
-            accounts,
-            solana_accounts,
-            trace_call_config,
-        )
-        .await
-    })
+    let tracer = Arc::new(RwLock::new(Some(Tracer::new(
+        trace_call_config.trace_config.enable_return_data,
+    ))));
+
+    let (emulation_result, _storage) = emulate_transaction(
+        rpc_client,
+        evm_loader,
+        tx,
+        token,
+        chain_id,
+        steps,
+        commitment,
+        accounts,
+        solana_accounts,
+        trace_call_config,
+        Some(tracer.clone()),
+    )
     .await?;
 
-    let (vm_trace, full_trace_data) = tracer.into_traces();
+    let (vm_trace, full_trace_data) = tracer.write().unwrap().take().unwrap().into_traces();
```

Comment:
I think we should handle unwraps and return Result instead.

Reply:
- The first `.unwrap()` is on a `Result<Guard, PoisonError<Guard>>`. `PoisonError` happens whenever a thread fails while the lock is held. I do not see how the thread holding the lock can fail while holding it.
- The second `.unwrap()` is on an `Option<Tracer>`. This can never fail because the tracer is defined as
```
    Some(Tracer::new(
        trace_call_config.trace_config.enable_return_data,
    ))
```
a few lines above in
```
    let tracer = Arc::new(RwLock::new(Some(Tracer::new(
        trace_call_config.trace_config.enable_return_data,
    ))));
```
This `Option<>` is needed because the function `pub fn into_traces(self)` consumes the `Tracer` value, so it needs to be `take()`n out of the `Option<>`.

Reply:
I agree with @andreisilviudragnea reasoning, especially about the poison error, but personally I would replace unwraps with `expect`.